src/inmanta/protocol/rest/__init__.py

```diff
@@ -47,59 +48,31 @@
 """
 
 
-def authorize_request(
-    auth_data: Optional[auth.claim_type], metadata: dict[str, str], message: JsonType, config: common.UrlMethod
-) -> None:
-    """
-    Authorize a request based on the given data
-    """
-    if auth_data is None:
-        return
-
-    # Enforce environment restrictions
-    env_key: str = const.INMANTA_URN + "env"
-    if env_key in auth_data:
-        if env_key not in metadata:
-            raise exceptions.Forbidden("The authorization token is scoped to a specific environment.")
-
-        if metadata[env_key] != "all" and auth_data[env_key] != metadata[env_key]:
-            raise exceptions.Forbidden("The authorization token is not valid for the requested environment.")
-
-    # Enforce client_types restrictions
-    ok: bool = False
-    ct_key: str = const.INMANTA_URN + "ct"
-    for ct in auth_data[ct_key]:
-        if ct in config.properties.client_types:
-            ok = True
-
-    if not ok:
-        raise exceptions.Forbidden(
-            "The authorization token does not have a valid client type for this call."
-            + f" ({auth_data[ct_key]} provided, {config.properties.client_types} expected"
-        )
-
-
 class CallArguments:
     """
     This class represents the call arguments for a method call.
     """
 
     def __init__(
-        self, properties: common.MethodProperties, message: dict[str, Optional[object]], request_headers: Mapping[str, str]
+        self, config: common.UrlMethod, message: dict[str, Optional[object]], request_headers: Mapping[str, str]
     ) -> None:
         """
-        :param method_config: The method configuration that contains the metadata and functions to call
-        :param message: The message recieved by the RPC call
+        :param config: The method configuration that contains the metadata and functions to call
+        :param message: The message received by the RPC call
         :param request_headers: The headers received by the RPC call
+        :param handler: The handler for the call
         """
-        self._properties = properties
+        self._config = config
+        self._properties = self._config.properties
         self._message = message
         self._request_headers = request_headers
         self._argspec: inspect.FullArgSpec = inspect.getfullargspec(self._properties.function)
 
         self._call_args: JsonType = {}
         self._headers: dict[str, str] = {}
         self._metadata: dict[str, object] = {}
```

Comment:
This one could use some documentation. As I understand this is for argument mappers to communicate with the authentication? and`const.INMANTA_URN + "env" ` and `const.INMANTA_URN + "ct"` are very important